### ISSUE TITLE
[Quick Question] in flash-attn  2.5.2, if I disable causal attn, does the attention_mask flash-attn 2 use remain the same or is it changed internally?

### ISSUE THREAD

**Issue by SStoica12** [Dao-AILab/flash-attention#2631]
Hello,

I found this https://github.com/Dao-AILab/flash-attention/issues/1304 in Oct 30, 2024 that said that when causal=False, is the attention mask an all-one matrix, so each query attends to all the key (complete bidirectionally). However, this was about 9 months since flash-attn  2.5.2, was released (Jan 31, 2024). 

The reason I ask is because I would like to try and use flash-attn v2.5.2 but have bidirectional attention on the image tokens and causal on the text tokens. I am working with the LLaVA 1.5 v2 from 2023.

Thank you!
